src/solver/pad_constant/pad_constant_fwd_contiguous.cpp

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/conv_solution.hpp"
+#include "miopen/datatype.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/kernel_build_params.hpp"
+#include "miopen/miopen.h"
+#include "miopen/mlo_internal.hpp"
+#include "miopen/pad_constant/solvers.hpp"
+#include "miopen/pad_constant/invoke_params.hpp"
+
+namespace miopen {
+namespace solver {
+namespace pad_constant_fwd_contiguous {
+bool PadConstantFwdContiguous::IsApplicable(
+    const ExecutionContext& /*context*/,
```

Comment:
I did the same thing before, and amd told me to use [[maybe_unused]].
```
[[maybe_unused]]const ExecutionContext& context,
```

Reply:
This one I am confused about, since `[[maybe_unused]]` is only supposed to be used if you are going to "maybe" use it later (eg. if you have an `#ifdef` that would make use of it). In this case, it is never going to be used at all.
C++ Core Guidelines have a topic on this, where they **explicitly** says that it should be unnamed, not marked `maybe_unused`: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-unused

Reply:
Currently, the use of [[maybe_unused]] in MIOpen is ambiguous, but o2buzzle's opinion seems reasonable.

Reply:
So are we going to keep it as-is or revert https://github.com/moreh-dev/MIOpen/pull/4/commits/4871cfd88549ae686fa5c427207f1506109f56d8 ?

Reply:
We are also PR on MIOpen as an external contributor. I think you might understand that we are making recommendations on these areas. Regarding this part, I think it would be a good idea to work on /*context*/ and then discuss it with AMD's reviewer.